plugins/manarefresher/refresher.go

```diff
@@ -0,0 +1,109 @@
+package manarefresher
+
+import (
+	"time"
+
+	"github.com/cockroachdb/errors"
+
+	"github.com/iotaledger/goshimmer/packages/clock"
+	"github.com/iotaledger/goshimmer/packages/ledgerstate"
+	"github.com/iotaledger/goshimmer/packages/tangle"
+	"github.com/iotaledger/goshimmer/plugins/autopeering/local"
+	"github.com/iotaledger/goshimmer/plugins/messagelayer"
+)
+
+// maxBookedAwaitTime is the time the refresher waits for an issued tx to become booked.
+const maxBookedAwaitTime = 5 * time.Second
+
+// Refresher is a component that takes care of refreshing the mana delegated to the node.
+type Refresher struct {
+	wallet   *wallet
+	receiver *DelegationReceiver
+}
+
+// NewRefresher creates a new Refresher object.
+func NewRefresher(wallet *wallet, receiver *DelegationReceiver) *Refresher {
+	return &Refresher{
+		wallet:   wallet,
+		receiver: receiver,
+	}
+}
+
+// Refresh scans the tangle for delegated outputs, and refreshes the node's mana by moving those.
+func (r *Refresher) Refresh() (err error) {
+	delegationOutputs := r.receiver.Scan()
+	if len(delegationOutputs) == 0 {
+		return
+	}
+	numberOfChunks := len(delegationOutputs)/ledgerstate.MaxInputCount + 1
```

Comment:
why the +1?

Reply:
`numberOfChunks` essentially defines how many transaction will we submit, because we can only submit 127 outputs in one tx.
So if there are less, than 127 outputs, len(outputs)/127 will be zero. (integer division) That's why you add +1.